rust/agama-lib/share/profile.schema.json

```diff
@@ -219,54 +221,222 @@
+              "enum": [ "delete", "resize", "keep" ]
+            },
+            "volumes": {
+              "title": "Set of volumes (file systems) to create",
+              "type": "array",
+              "items": {
+                "type": "object",
+                "additionalProperties": false,
+                "required": ["mount"],
+                "properties": {
+                  "mount": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "required": ["path"],
+                    "properties": {
+                      "path": {
+                      "title": "Mount path",
+                      "type": "string"
+                      },
+                      "options": {
+                        "title": "Options to add to the fourth field of fstab",
+                        "type": "array",
+                        "items": { "type": "string" }
+                      }
+                    }
+                  },
+                  "filesystem": {
+                    "title": "File system of the volume",
+                    "type": ["string", "object"],
+                    "additionalProperties": false,
+                    "properties": {
+                      "btrfs": {
+                        "title": "Specification of a Btrfs file system",
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": {
+                          "snapshots": {
+                            "title": "Whether Btrfs snapshots should be configured",
+                            "type": "boolean"
+                          }
+                        }
+                      }
+                    }
+                  },
+                  "size": {
+                    "title": "Size limits",
+                    "description": "It can be 'auto', a fixed size (as string or integer) or a set of size limits",
+                    "type": ["string", "integer", "array", "object"],
+                    "prefixItems": [
+                      {
+                        "title": "Mandatory lower size limit",
+                        "type": ["integer", "string"]
+                      },
+                      { 
+                        "title": "Optional upper size limit",
+                        "type": ["integer", "string"]
+                      }
+                    ],
+                    "additionalProperties": false,
+                    "required": ["min"],
+                    "properties": {
+                      "min": {
+                        "title": "Mandatory lower size limit",
+                        "type": ["integer", "string"]
+                      },
+                      "max": {
+                        "title": "Optional upper size limit",
+                        "type": ["integer", "string"]
+                      }
+                    }
+                  },
```

Comment:
```suggestion
                  "size": {
                    "title": "Size limits",
                    "description": "It can be 'auto', a fixed size (as string or integer) or a set of size limits",
                    "oneOf": [
                      { "const": "auto" },
                      { "$ref": "#/$defs/units" },
                      { "$ref": "#/$defs/bytes" },
                      {
                        "type": "array",
                        "items": { "#/$defs/value" },
                        "minItems": 1,
                        "minItems": 2
                      },
                      {
                        "type": "object",
                        "additionalProperties": false,
                        "required": ["min"],
                        "properties": {
                          "min": {
                            "title": "Mandatory lower size limit",
                            "type": { "#/$defs/value" }
                          },
                          "max": {
                            "title": "Optional upper size limit",
                            "type": { "#/$defs/value" }
                          }
                        }
                      }
                    ],
                    "$defs": {
                      "units": {
                        "type": "string",
                        "pattern": "^\d+(\.\d+)?\s*(GiB|MiB|Kib)"
                      },
                      "bytes": {
                        "type": "integer",
                        "minimum": 0
                      },
                      "value": {
                        "oneOf": [
                          { "$ref": "#/$defs/units" },
                          { "$ref": "#/$defs/bytes" }
                        ]
                      }
                    }
                  },
```